raphtory/src/db/api/view/graph.rs

```diff
@@ -227,35 +227,35 @@ impl<'graph, G: BoxableGraphView + Sized + Clone + 'graph> GraphViewOps<'graph>
                 for (index, node) in self.nodes().iter().enumerate() {
                     let new_id = VID(index);
                     let gid = node.id();
-                    if let Some(new_node) = shard.set(new_id, gid.as_ref()) {
+                    if let Some(mut new_node) = shard.set(new_id, gid.as_ref()) {
                         node_map_shared[node.node.index()].store(index, Ordering::Relaxed);
                         if let Some(node_type) = node.node_type() {
                             let new_type_id = g
                                 .node_meta
                                 .node_type_meta()
                                 .get_or_create_id(&node_type)
                                 .inner();
-                            new_node.node_type = new_type_id;
+                            new_node.node_store_mut().node_type = new_type_id;
                         }
                         g.logical_to_physical.set(gid.as_ref(), new_id)?;
 
                         if let Some(earliest) = node.earliest_time() {
                             // explicitly add node earliest_time to handle PersistentGraph
-                            new_node.update_time(TimeIndexEntry::start(earliest))
+                            new_node
+                                .node_store_mut()
+                                .update_t_prop_time(TimeIndexEntry::start(earliest), None)
                         }
-                        for t in node.history() {
-                            new_node.update_time(TimeIndexEntry::start(t));
-                        }
-                        for t_prop_id in node.temporal_prop_ids() {
-                            for (t, prop_value) in
-                                self.temporal_node_prop_hist(node.node, t_prop_id)
-                            {
-                                new_node.add_prop(t, t_prop_id, prop_value)?;
-                            }
+
+                        for (t, rows) in node.rows() {
+                            let prop_offset = new_node.t_props_log_mut().push(rows)?;
+                            new_node.node_store_mut().update_t_prop_time(t, prop_offset);
```

Comment:
does this handle the property value at the start of the window correctly for a windowed PersistentGraph? Do we have a test for that?